File: Programmers/정렬_가장큰수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool sorting(int a, int b){
    string a_s = to_string(a);
    string b_s = to_string(b);
    int sum1 = stoi(a_s+b_s);
    int sum2 = stoi(b_s+a_s);
    if(sum1 > sum2) return true;
    else return false;
}

string solution(vector<int> numbers) {
    string answer = "";
    
    sort(numbers.begin(), numbers.end(), sorting);
    
    for(int n: numbers){
        string n_s = to_string(n);
        answer += n_s;
    }
    answer = numbers.front() == 0 ? "0" : answer;
    return answer;
}

File: Programmers/정렬_가장큰수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] str = new String[numbers.length];
        for(int i=0;i<str.length;i++){
            str[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(str, new Comparator<String>(){
            public int compare(String str1, String str2){
                return (str2+str1).compareTo(str1+str2);
            }
        });
        
        if(str[0].equals("0")){
            return "0";
        }
        
        for(String s:str){
            answer += s;
        }
        
        return answer;
    }
}